tool_collections/bamtools/bamtools_split/bamtools-split.xml

```diff
@@ -1,108 +1,146 @@
+                ln -s "${input_bam.metadata.bam_index}" "localbam_${bam_count}.bam.bai" &&
+            #end for
+            bamtools
+            split
+            #if str ( $analysis_type.analysis_type_selector ) == "-tag" :
+                ${analysis_type.analysis_type_selector} "${analysis_type.tag_name}"
+            #else
+                ${analysis_type.analysis_type_selector}
+            #end if
+            -stub split_bam
+            #for $bam_count, $input_bam in enumerate( $input_bams ):
+                -in "localbam_${bam_count}.bam"
+            #end for
+            #if str ( $analysis_type.analysis_type_selector ) == "-reference"  and $analysis_type.output_type == "dataset_collection": 
+                #import re
+                #set $name = $re.sub('\W','_',$re.sub('\.bam$','',$input_bams[0].name))
+                #set $ref_list = ' '.join([$re.sub('^.*__sq__(.+)__sq__.*$','\\1',n) if n.find('__sq__') >= 0 else n for n in str($input_bam.metadata.reference_names).split(',')])
+                    && (export I=0; 
+                      for i in $ref_list; 
+                        do I=\$((++I)); SN=`printf "split_bam.REF_%s.bam" "\$i"`; 
+                           if [ -e \$SN ]; 
+                             then FN=`printf "split_bam%05d%s.%s.bam" \$((I)) "$name" "\$i"`; 
+                             mv \$SN \$FN;
+                           fi;
+                        done)
+            #end if
+        ]]>
+    </command>
+    <inputs>
+        <param name="input_bams" type="data" format="bam" label="BAM dataset(s) to filter" min="1" multiple="True"/>
+        <conditional name="analysis_type">
+            <param name="analysis_type_selector" type="select" label="Split BAM dataset(s) by" help="See help below for explanation of each option">
+                <option value="-mapped">Mapping status (-mapped)</option>
+                <option value="-paired">Pairing status (-paired)</option>
+                <option value="-reference">Reference name (-reference)</option>
+                <option value="-tag">Specific tag (-tag)</option>
+            </param>
+            <when value="-mapped" />
+            <when value="-paired" />
+            <when value="-reference" >
+                <expand macro="macro_output_type" />
+            </when>
+            <when value="-tag">
+                <param name="tag_name" type="text" value="NM" label="Enter tag name here" help="For example, to split on NM tag enter &quot;NM&quot;"/>
+                <expand macro="macro_output_type" />
+            </when>
+        </conditional>
+    </inputs>
+    <outputs>
+        <data format="txt" name="report" label="BAMSplitter Run" hidden="true">
+            <discover_datasets pattern="split_bam\.(?P&lt;designation&gt;.+)\.bam" ext="bam" visible="true"/>
+            <filter>analysis_type['analysis_type_selector'] in ('-mapped','-paired') or analysis_type['output_type'] != 'dataset_collection'</filter>
```

Comment:
Discovered datsets just make little sense in the case of paired and mapping status - it should be two separate outputs. That seems like a useful operation in workflows for instance but we cannot use this tool in workflows in these cases.

I'd split this one tool into three Galaxy tools with fixed output types:

- Bam Split on reference or tag - always produces a dataset list.
- Bam Split paired - always produces a dataset pair.
- Bam Split on mapping status - always produces two datasets.

Ping @nekrut